odl/discr/discr_mappings.py

```diff
@@ -467,219 +482,307 @@ def _call(self, x, out=None):
             ``out`` was provided, the returned object is a reference
             to it.
         """
+        # TODO: pass reasonable options on to the interpolator
         def linear(arg, out=None):
             """Interpolating function with vectorization."""
             if is_valid_input_meshgrid(arg, self.grid.ndim):
-                # TODO: check if this works for 'F' ordering
-                interp = _LinearMeshgridInterpolator(
+                interp = _LinearInterpolator(
                     self.grid.coord_vectors,
-                    x.data.reshape(self.grid.shape, order=self.order))
+                    x.data.reshape(self.grid.shape, order=self.order),
+                    typ='meshgrid')
             else:
-                # TODO: check if points are ordered. If not, use
-                # LinearNDInterpolator
-                interp = _LinearPointwiseInterpolator(
+                interp = _LinearInterpolator(
                     self.grid.coord_vectors,
-                    x.data.reshape(self.grid.shape, order=self.order))
+                    x.data.reshape(self.grid.shape, order=self.order),
+                    typ='array')
+
             return interp(arg, out=out)
 
         return self.range.element(linear, vectorized=True)
 
 
-class _PointwiseInterpolator(object):
+class PerAxisInterpolation(FunctionSetMapping):
+
+    """Interpolation scheme set for each axis individually."""
+
+    def __init__(self, ip_fspace, grid, dspace, schemes,
+                 nn_variants=None, order='C'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        fspace : `FunctionSpace`
+            The undiscretized (abstract) space of functions to be
+            discretized. Its field must be the same as that of data
+            space. Its `Operator.domain` must be an
+            `IntervalProd`.
+        grid :  `TensorGrid`
+            The grid on which to evaluate. Must be contained in
+            the domain of the function set.
+        dspace : `FnBase`
+            Data space providing containers for the values of a
+            discretized object. Its size must be equal to the
+            total number of grid points. Its field must be the same
+            as that of the function space.
+        schemes : `str` or `list` of `str`
+            Indicates which interpolation scheme to use for which axis.
+            A single string is interpreted as a global scheme for all
+            axes.
+        nn_variants : `str` or `list` of `str`, optional
+            Which variant ('left' or 'right') to use in nearest neighbor
```

Comment:
Since the use of this kw depends on the choice of `schemes` this could be rendered unused. I'd thus make it part of `kwargs`, or at the least put it last.